modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/object/entries/frontend/data/set/test/RelatedModelFDSActionProviderTest.java

```diff
@@ -0,0 +1,201 @@
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.mock.web.MockHttpServletRequest;
+
+/**
+ * @author Feliphe Marinho
+ */
+@RunWith(Arquillian.class)
+public class RelatedModelFDSActionProviderTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Test
+	public void testGetDropdownItems() throws Exception {
+		ObjectDefinition objectDefinition =
+			_objectDefinitionLocalService.addCustomObjectDefinition(
+				TestPropsValues.getUserId(), 0, null, false, false, true, true,
+				false,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				ObjectDefinitionTestUtil.getRandomName(), null, null,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				true, ObjectDefinitionConstants.SCOPE_COMPANY,
+				ObjectDefinitionConstants.STORAGE_TYPE_DEFAULT,
+				Arrays.asList(
+					new TextObjectFieldBuilder(
+					).labelMap(
+						LocalizedMapUtil.getLocalizedMap(
+							RandomTestUtil.randomString())
+					).name(
+						"textObjectFieldName"
+					).build()));
+
+		objectDefinition =
+			_objectDefinitionLocalService.publishCustomObjectDefinition(
+				TestPropsValues.getUserId(),
+				objectDefinition.getObjectDefinitionId());
+
+		ObjectEntry objectEntry = _objectEntryLocalService.addObjectEntry(
+			TestPropsValues.getUserId(), 0,
+			objectDefinition.getObjectDefinitionId(), Collections.emptyMap(),
+			ServiceContextTestUtil.getServiceContext());
+
+		Role role = RoleTestUtil.addRole(RoleConstants.TYPE_REGULAR);
+
+		_resourcePermissionLocalService.setResourcePermissions(
+			TestPropsValues.getCompanyId(), objectDefinition.getClassName(),
+			ResourceConstants.SCOPE_COMPANY,
+			String.valueOf(TestPropsValues.getCompanyId()), role.getRoleId(),
+			new String[] {ActionKeys.UPDATE});
+
+		User user = UserTestUtil.addUser();
+
+		_userLocalService.addRoleUser(role.getRoleId(), user);
+
+		PermissionThreadLocal.setPermissionChecker(
+			PermissionCheckerFactoryUtil.create(user));
+
+		PrincipalThreadLocal.setName(user.getUserId());
+
+		List<DropdownItem> dropdownItems = _fdsActionProvider.getDropdownItems(
+			TestPropsValues.getGroupId(),
+			_getMockHttpServletRequest(objectEntry.getObjectEntryId()),
+			new RelatedModel(
+				objectDefinition.getClassName(), objectEntry.getObjectEntryId(),
+				objectEntry.getTitleValue(), false));
+
+		Assert.assertEquals(dropdownItems.toString(), 2, dropdownItems.size());
+
+		DropdownItem dropdownItem1 = dropdownItems.get(0);
+
+		Assert.assertEquals("view", dropdownItem1.get("label"));
+
+		DropdownItem dropdownItem2 = dropdownItems.get(1);
+
+		Assert.assertEquals("delete", dropdownItem2.get("label"));
```

Comment:
This code repeats twice here and one more later. What about making a private method like:
```
private void _assertDropdownItems(
		List<DropdownItem> dropdownItems, String[] expectedValues) {

          for (int i = 0; i < dropdownItems.size(); i++) {
	          DropdownItem dropdownItem = dropdownItems.get(i);
          
	          Assert.assertEquals(expectedValues[i], dropdownItem.get("label"));
          }
}
```

Reply:
Done [here](https://github.com/brianchandotcom/liferay-portal/pull/158416/commits/13099bdbc5fd711dc5a18a343f79506e32182183)